cpp/src/neighbors/detail/knn_brute_force.cuh

```diff
@@ -81,12 +81,13 @@ void tiled_brute_force_knn(const raft::resources& handle,
                            const uint32_t* filter_bitmap             = nullptr)
 {
   // Figure out the number of rows/cols to tile for
-  size_t tile_rows   = 0;
-  size_t tile_cols   = 0;
-  auto stream        = raft::resource::get_cuda_stream(handle);
-  auto device_memory = raft::resource::get_workspace_resource(handle);
-  auto total_mem     = rmm::available_device_memory().second;
+  size_t tile_rows = 0;
+  size_t tile_cols = 0;
+  auto stream      = raft::resource::get_cuda_stream(handle);
 
+  // total memory is not relevant in the heuristic for data below 512 MB
+  auto total_mem =
+    (sizeof(DistanceT) * m * n < 1 << 29) ? (1ul << 36) : rmm::available_device_memory().second;
```

Comment:
Rather than tweak the total memory being passed to `chooseTileSize` - can we just adjust the chooseTileSize function instead to handle smaller inputs properly?  This will consolidate all the logic into a single function, rather than requiring some of the logic to be handled by the caller of chooseTileSize